src/main/java/org/ow2/proactive/connector/iaas/cloud/provider/maas/MaasProvider.java

```diff
@@ -0,0 +1,248 @@
+                                      .collect(Collectors.toSet());
+    }
+
+    @Override
+    /**
+     * Upload a new *decommissioning* script to MAAS region controller.
+     */
+    public List<ScriptResult> executeScriptOnInstanceId(Infrastructure infrastructure, String instanceId,
+            InstanceScript instanceScript) {
+
+        ScriptResult scriptResult = new ScriptResult(instanceId, "", "");
+
+        StringBuilder script = new StringBuilder();
+        script.append("#!/bin/bash\n");
+        for (int i = 0; i < instanceScript.getScripts().length; i++) {
+            script.append("\n");
+            script.append(instanceScript.getScripts()[i]);
+        }
+
+        CommissioningScript maasScript = maasProviderClientCache.getMaasClient(infrastructure)
+                                                                .postCommissioningScript(script.toString().getBytes(),
+                                                                                         instanceId);
+
+        if (maasScript == null) {
+            return Lists.newArrayList(scriptResult.withError("Unable to upload script " + instanceId));
+        }
+
+        // Unable to retrieve scripts output, returns empty results instead
+        return IntStream.rangeClosed(1, instanceScript.getScripts().length)
+                        .mapToObj(scriptNumber -> new ScriptResult(instanceId, "", ""))
+                        .collect(Collectors.toList());
+    }
+
+    @Override
+    public List<ScriptResult> executeScriptOnInstanceTag(Infrastructure infrastructure, String instanceTag,
+            InstanceScript instanceScript) {
+
+        return executeScriptOnInstanceId(infrastructure,
+                                         maasProviderClientCache.getMaasClient(infrastructure)
+                                                                .getMachines()
+                                                                .stream()
+                                                                .filter(machine -> machine.getHostname()
+                                                                                          .equals(instanceTag))
+                                                                .findFirst()
+                                                                .orElseThrow(() -> new RuntimeException("ERROR machine with hostname '" +
+                                                                                                        instanceTag +
+                                                                                                        "' not found"))
+                                                                .getSystemId(),
+                                         instanceScript);
+    }
+
+    @Override
+    public Set<Image> getAllImages(Infrastructure infrastructure) {
+        throw new NotSupportedException("Operation not supported for MAAS");
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "Operation not supported for MAAS" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1192)